packages/react-ui-ag/src/Carousels/ListingCarousel.js

```diff
@@ -10,7 +10,7 @@ import {
 } from '../Listings'
 
 @themed(/^ListingCarousel/)
```

Comment:
Make sure you add a `{ pure: true }` to the themed arguments

Reply:
Also, do you still need this check: `componentWillReceiveProps(nextProps) {
    if (nextProps.selectedIndex !== this.props.selectedIndex) {
      this.updateSelectedIndex(nextProps.selectedIndex)
    }
  }`
Since this is now a pure component?

Reply:
Yes. The only difference between `PureComponent` and `Component` is that `shouldComponentUpdate` is done for you and is a shallow comparison of all props. So basically by changing this to a `PureComponent` he's ensuring that props are compared shallowly rather than not at all.